rts/motoko-rts/src/gc/incremental/phases/update_increment.rs

```diff
@@ -102,7 +102,12 @@ impl<'a> UpdateIncrement<'a> {
             // Keep mark bit and later resume updating more slices of this array.
             return;
         }
+        #[enhanced_orthogonal_persistence]
+        debug_assert!(object.tag() < TAG_ARRAY_SLICE_MIN || object.tag() == TAG_WEAK_REF);
+
+        #[classical_persistence]
```

Comment:
`TAG_ARRAY_SLICE_MIN` would need to be adjusted to cover `TAG_WEAK_REF` (left a comment below).
```suggestion
```